app.py

```diff
@@ -1,6 +1,59 @@
-from flask import Flask
-from models import db, connect_db
+import os
+
+from flask import Flask, session, g, render_template, redirect, flash
+from forms import RegisterUser
+from models import db, connect_db, User
+from sqlalchemy.exc import IntegrityError
+
+
+CURR_USER_KEY = "curr_user"
 
 app = Flask(__name__)
+app.config['SECRET_KEY'] = os.environ.get("SECRET")
 
 connect_db(app)
+
+@app.before_request
+def add_user_to_g():
+    """If user has logged in, add current user to Flask global object."""
+    if CURR_USER_KEY in session:
+        g.user = User.query.get(session[CURR_USER_KEY])
+    else:
+        g.user = None
+
+def do_login(user):
+    """Method logs in the user."""
+    session[CURR_USER_KEY] = user.id
+
+
+@app.route('/register', methods=["GET", "POST"])
+def register_user():
+    """Handles user signup."""
+    form = RegisterUser()
+
+    if form.validate_on_submit():
+        try:
+            user = User.signup(
+                firstname=form.firstname.data,
+                lastname=form.lastname.data,
+                username=form.username.data,
+                email=form.email.data,
+                password=form.password.data,
+                image=form.image.data,
+                state=form.state.data,
+                vax_date=form.vax_date.data,
+                covid_status=form.covid_status.data)
+            db.session.commit()
+        except IntegrityError:
+            flash("Invalid information", 'danger')
+            return render_template('users/register.html', form=form)
+        do_login(user)
+        return redirect('/user')
```

Comment:
@azaria-dedmon Isn't this supposed to be in the `try` block ? What happens when a user signs up successfully ?

Reply:
I am using the try block to test specifically on whether or not the user is able to actually register for an account. I still have the action of the user's redirect nested inside of the if statement. Would it be considered best practice to include the redirect inside of the try block as well?

Reply:
I guess if the user successfully registers, the `do_login` function is called then a redirect occurs. If that is the case, then the `do_login` function should be in the `try` block.